xdsl/irdl.py

```diff
@@ -1072,12 +1074,26 @@ def irdl_op_definition(cls: type[_OpT]) -> type[_OpT]:
             new_attrs[attribute_name] = property(
                 lambda self, name=attribute_name: self.attributes[name])
 
-    def builder(cls,
-                operands=[],
-                result_types=[],
-                attributes=dict(),
-                successors=[],
-                regions=[]):
+    def builder(
+        cls: type[_OpT],
+        operands: list[SSAValue | Operation
```

Comment:
This now reflects the type signature of the function being overloaded. In main, passing `None` triggers no type errors, crashing at runtime instead